lib/fs.js

```diff
@@ -535,9 +545,17 @@ function tryReadSync(fd, isUserFd, buffer, pos, len) {
 }
 
 fs.readFileSync = function(path, options) {
-  options = getOptions(options, { flag: 'r' });
+  var flag;
+  var encoding;
+  if (options != null) {
+    options = getOptions(options, defaultReadFileOpts);
```

Comment:
Lots of duplicates, you have O_RDONLY in defaultReadFileOpts and in 552 and 555 line. Better to remove `defaultReadFileOpts` and add `defaltReadFileFlag`. `stringToFlags` should be in getOptions.
```
const defaltReadFileFlag = O_RDONLY;
```
```
if (options != null) {
    options = getOptions(options);
    flag = options.flag || defaultReadFileOpts;
    encoding = options.encoding;
} else {
    flag = defaultReadFileOpts;
}
```

Reply:
And `getOptions` should be like:
```
function getOptions(options) {
  const type = typeof options;
  if (type === 'string') {
    return { flag: stringToFlags(options) };
  } else if (type !== 'object') {
    throw new TypeError('"options" must be a string or an object, got ' +
                        typeof options + ' instead.');
  }
  if (options.flag)
    options.flag = stringToFlags(options);
  if (options.encoding !== 'buffer')
    assertEncoding(options.encoding);
  return options;
}
```